backend/main.py

```
@@ -310,6 +311,31 @@ async def stripe_webhook(request: Request, db: Session = Depends(get_db)):
+# anyone can type ?payment=success into the URL. This endpoint asks Stripe
+# directly whether the given session actually completed, so the frontend can
+# safely grant on-device access only after a trusted answer.
+
+@app.get("/checkout/verify-session")
+@limiter.limit("30/minute")
+async def verify_checkout_session(request: Request, session_id: str):
+    if not STRIPE_SECRET_KEY:
+        raise HTTPException(500, "Stripe not configured")
+    try:
+        session = stripe.checkout.Session.retrieve(session_id)
+    except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)